### DISCUSSION TITLE
How to manually pad a table cell with whitespace / avoid removal of whitespace?

### DISCUSSION THREAD

**Discussion by bluenote10** [Textualize/rich#3153] [Q&A]
I'd like to combine `justify="right"` with a manual whitespace padding to achieve the desired alignment result. Consider for instance:

```py
from rich.console import Console
from rich.table import Table

table = Table()
table.add_column("Change", justify="right")

table.add_row("2.1 % lower ")
table.add_row("3.2 % higher")
table.add_row("3.0 % higher")
table.add_row("5.2 % lower ")
table.add_row("9.1 % higher")

console = Console()
console.print(table)
```

Note that I've added a whitespace after "lower" so that it should match in width with "higher". Unfortunately, this whitespace seems to get removed implicitly. The table looks like:

```
┏━━━━━━━━━━━━━━┓
┃       Change ┃
┡━━━━━━━━━━━━━━┩
│  2.1 % lower │
│ 3.2 % higher │
│ 3.0 % higher │
│  5.2 % lower │
│ 9.1 % higher │
└──────────────┘
```

I.e., the numbers don't align on the `.` because my manual padding got removed.

Is there a trick to avoid the removal of the padding?

**Comment by willmcgugan** [comment]
If you have added spaces, you have essentially justified the cell contents yourself. Remove the justify form the column and your decimal points will line up...

**Reply by bluenote10** [reply]
> If you have added spaces, you have essentially justified the cell contents yourself. Remove the justify form the column and your decimal points will line up...

I don't really understand. The will just left-align them so that aren't aligned at all. For instance:

```py
from rich.console import Console
from rich.table import Table

table = Table()
table.add_column("Change")

table.add_row("2.1 % lower")
table.add_row("36.2 % higher")
table.add_row("3.0 % higher")
table.add_row("523.2 % lower")
table.add_row("90.1 % higher")

console = Console()
console.print(table)
```

Results in a messy table:

```
┏━━━━━━━━━━━━━━━┓
┃ Change        ┃
┡━━━━━━━━━━━━━━━┩
│ 2.1 % lower   │
│ 36.2 % higher │
│ 3.0 % higher  │
│ 523.2 % lower │
│ 90.1 % higher │
└───────────────┘
```

But I'd like:

```
┏━━━━━━━━━━━━━━━━┓
┃         Change ┃
┡━━━━━━━━━━━━━━━━┩
│   2.1 % lower  │
│  36.2 % higher │
│   3.0 % higher │
│ 523.2 % lower  │
│  90.1 % higher │
└────────────────┘
```


I.e., I'd like to right-align the elements, because they are inherently numbers with an arbitrary number of leading digital places, where right alignment is best -- just with a fixed width text suffix.
